service/src/utils/health-check.ts

```diff
@@ -1,44 +1,83 @@
-    const freeBond = await web3Interface.viewStateFacet.getFreeBond(
-      web3Interface.ethWallet.address
-    );
-    const ethBal = await web3Interface.signer.getBalance();
+    const profile = await fetchProfileOrPromptProfileSetup(web3Interface);
 
-    console.log(
-      "\n\n=========================================================================================================\n"
-    );
-    console.log(" YOUR BALANCES:\n");
+    const sarcoBalance = await getSarcoBalance(web3Interface);
+    const ethBalance = await getEthBalance(web3Interface);
+    const freeBondBalance = await getFreeBondBalance(web3Interface);
 
-    console.log(` * SARCO:             ${ethers.utils.formatEther(bal)} SARCO`);
-    console.log(` * ETHER:             ${ethers.utils.formatEther(ethBal)} ETH`);
-    if (ethBal.lte(ethers.utils.parseEther("0.0005"))) {
-      archLogger.warn(
-        `\n   You have very little ETH in your account. You may not be able to sign any transactions (or do unwrappings)!\n`
-      );
-    }
+    logCallout(async () => {
+      logBalances(sarcoBalance, ethBalance, freeBondBalance);
 
-    console.log(` * Free Bond:         ${ethers.utils.formatEther(freeBond)} SARCO`);
-    if (freeBond.eq(constants.Zero)) {
-      archLogger.warn(`\n   You have no free bond. You will not be able to accept new jobs!`);
-      archLogger.error(
-        `   Run: \`npm run start -- --deposit-bond:<amount>\` to deposit some SARCO\n`
-      );
-    }
-    console.log(
-      "\n=========================================================================================================\n\n"
-    );
+      // If ETH balance is low, the archaeologist won't have gas to sign transactions
+      warnIfEthBalanceIsLow(ethBalance);
+
+      // Free bond must be >= their min digging fee to accept new jobs
+      await warnIfFreeBondIsLessThanMinDiggingFee(freeBondBalance, profile.minimumDiggingFee);
+    });
   } catch (e) {
     archLogger.error(e);
     exit(RPC_EXCEPTION);
   }
 }
+
+const fetchProfileOrPromptProfileSetup = async (
+  web3Interface: Web3Interface
+): Promise<OnchainProfile> => {
+  const profile = await getOnchainProfile(web3Interface);
+  if (!profile.exists) {
+    logCallout(() => {
+      console.log(" ARCHAEOLOGIST NOT REGISTERED:\n");
+
+      archLogger.warn(`\n   Your archaeologist is not yet registered.`);
+      archLogger.error(
+        `   Run: \`npm run register -- --deposit-bond:<amount>\` to register your archaeologist with some free bond\n`
+      );
+    });
+
+    exit(NO_ONCHAIN_PROFILE);
+  }
+
+  return profile;
+};
+
+const warnIfFreeBondIsLessThanMinDiggingFee = (
+  freeBondBal: BigNumber,
+  minDiggingFee: BigNumber
+): void => {
+  if (freeBondBal.lt(minDiggingFee)) {
+    archLogger.warn(
+      `\n   Your free bond is less than your minimum digging fee. You will not be able to accept new jobs!`
+    );
+    archLogger.error(
+      `   Run: \`npm run start -- --deposit-bond:<amount>\` to deposit some SARCO\n`
```

Comment:
Deposit bond needs to be updated to the new CLI -- this will be done in a followup task